quadcontrol/scenario_parser.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Phoenix-Pilot
+ *
+ * Flight scenario parser
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include "scenario_parser.h"
+#include <parser.h>
+
+#include <stdlib.h>
+
+
+#define INIT_SIZE 5
+
+
+static int invCnt;
+
+static struct {
+	flight_mode_t *result;
+	int invCnt; /* number of converters invocations */
+} scenarioParser_common;
```

Comment:
While data is parsed, you need to keep the pointer to data and size in a global variable. I suggest to make an array for each parser. 
This solution would not be thread safe.

```suggestion
enum { config_scenarioID = 0, config_pidID, config_end  };

static struct {
	void *result;
	size_t sz;  /* number of parsed data */
} res[config_end];
```